#96 Include tests in CI Jenkins Build and fix build

ndoschek opened the pull request:
- Add code linting as separate pipeline step
- Add test execution as separate pipeline step
- Add clean of yarn cache to fix sporadic build errors
- Add GitHub back reporting for ESLint and Test execution
- Do not execute integration on Jenkins (connection issues with localhost)

Fixes #96

cdamus reviewed (changes requested):
LGTM! And the various new build options seem to work locally (I don't have an environment for testing the Jenkins configuration).

One thing: there seem to be a lot of left-overs in the `yarn.lock`. When I ran a `yarn` build, a lot of dependencies in the lockfile were removed and several more were upversioned. Maybe include a rebuild of the lock file in this PR?

ndoschek commented:
Thanks for catching that @cdamus! I added the rebuilt `yarn.lock` file and also added the `eslint.xml` to the `.gitignore` as for the other related PR. Thanks again!

cdamus reviewed (approved):
Thanks! That looks great.
